File: MusicShop.Model/Requests/GuitarUpdateRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MusicShop.Model.Requests
{
    public class GuitarUpdateRequest
    {
        public int? GuitarTypeId { get; set; }
        public string? Model { get; set; }
        public string? Pickups { get; set; }
        public string? PickupConfiguration { get; set; }
        public int? Frets { get; set; }

        public int? BrandId { get; set; }

        public decimal? Price { get; set; }
        public string? Description { get; set; }
        public DateTime? UpdatedAt { get; set; }
    }
}

File: UI/musicshop_mobile/lib/models/guitar/guitar_update_request.dart
import 'package:json_annotation/json_annotation.dart';

part 'guitar_update_request.g.dart';

@JsonSerializable()
class GuitarUpdateRequest {
  int? id;
  int? guitarTypeId;
  int? brandId;
  String? pickups;
  String? pickupConfiguration;
  String? description;
  String? model;
  int? frets;
  double? price;
  String? productImage;

  GuitarUpdateRequest({
    this.id,
  });

  factory GuitarUpdateRequest.fromJson(Map<String, dynamic> json) =>
      _$GuitarUpdateRequestFromJson(json);

  Map<String, dynamic> toJson() => _$GuitarUpdateRequestToJson(this);
}
